hathor/cli/run_node.py

```diff
@@ -164,24 +171,34 @@ def create_wallet():
                 raise ValueError('Invalid type for wallet')
 
         tx_storage: TransactionStorage
-        if args.data:
-            if args.rocksdb_storage:
-                tx_storage = TransactionRocksDBStorage(path=args.data, with_index=(not args.cache))
-            else:
-                tx_storage = TransactionCompactStorage(path=args.data, with_index=(not args.cache))
-            self.log.info('with storage', storage_class=type(tx_storage).__name__, path=args.data)
-            if args.cache:
-                tx_storage = TransactionCacheStorage(tx_storage, reactor)
-                if args.cache_size:
-                    tx_storage.capacity = args.cache_size
-                if args.cache_interval:
-                    tx_storage.interval = args.cache_interval
-                self.log.info('with cache', capacity=tx_storage.capacity, interval=tx_storage.interval)
-                tx_storage.start()
-        else:
+        if args.memory_storage:
+            assert not args.data, '--data should not be used with --memory-storage'
             # if using MemoryStorage, no need to have cache
             tx_storage = TransactionMemoryStorage()
             self.log.info('with storage', storage_class=type(tx_storage).__name__)
+        elif args.json_storage:
+            assert args.data, '--data is expected'
+            tx_storage = TransactionCompactStorage(path=args.data, with_index=(not args.cache))
+        elif args.old_rocksdb_storage:
+            assert args.data, '--data is expected'
+            self.log.warn('the old rocksdb storage is deprecated and support will be removed')
+            tx_storage = TransactionOldRocksDBStorage(path=args.data)
+        else:
+            assert args.data, '--data is expected'
+            if args.rocksdb_storage:
+                self.log.warn('--rocksdb-storage is now implied, no need to specify it')
+            cache_capacity = args.rocksdb_cache or None
```

Comment:
Why `or None`? You can set `None` as default in the argparse.

Reply:
Fixed.